### PR TITLE
feat: better state hasher wiring

### LINKED ISSUE #13208: Wiring improvements for the state hasher
Clean up the state hasher wiring.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #13208

### CODE REVIEW

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java [hunk 1]
```diff
@@ -745,7 +751,7 @@ public void bind(
         latestImmutableStateNexusWiring.bind(latestImmutableStateNexus);
         latestCompleteStateNexusWiring.bind(latestCompleteStateNexus);
         savedStateControllerWiring.bind(savedStateController);
-        signedStateHasherWiring.bind(signedStateHasher);
+        stateHasherWiring.bind(builder.buildStateHasher()); // TODO
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java:754]: todo

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java [hunk 2]
```diff
@@ -892,4 +895,35 @@ public Gossip buildGossip() {
...
+
+    /**
+     * Build the state hasher if it has not yet been built. If one has been provided via
+     * {@link #withStateHasher(StateHasher)}, that hasher will be used. If this method is called more than once, only
+     * the first call will build the state hasher. Otherwise, the default hasher will be created and returned.
+     *
+     * @return the signed state hasher
+     */
+    @NonNull
+    public StateHasher buildStateHasher() {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:923]: For the `PlatformComponentBuilder` is a builder for `Platform`:

```
Platform platform = new PlatformComponentBuilder(foo).withStatusStateMachine(bar).build();
```

I do not see why this method is `public`?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java [hunk 3]
```diff
@@ -114,8 +113,10 @@ public record PlatformSchedulersConfig(
         @ConfigProperty(defaultValue = "SEQUENTIAL_THREAD") TaskSchedulerType hashLoggerSchedulerType,
         @ConfigProperty(defaultValue = "100") int hashLoggerUnhandledTaskCapacity,
         @ConfigProperty(defaultValue = "1000") int completeStateNotifierUnhandledCapacity,
-        @ConfigProperty(defaultValue = "SEQUENTIAL_THREAD") TaskSchedulerType stateHasherSchedulerType,
-        @ConfigProperty(defaultValue = "2") int stateHasherUnhandledCapacity,
+        @ConfigProperty(
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java:116]: Creates https://github.com/hashgraph/hedera-services/issues/13262 
Maybe we can improve the config here.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/generate-platform-diagram.sh [hunk 4]
```diff
@@ -12,6 +12,7 @@ pcli diagram \
     -l 'TransactionPrehandler:futures:consensusRoundHandler' \
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/generate-platform-diagram.sh:12]: I think we need to move that file. I created https://github.com/hashgraph/hedera-services/issues/13263

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:923]: Should not be public, but currently our package layout doesn't allow non-public methods. I agree that the way you suggest is the correct way, but first we need to reorganize our package structure.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java:116]: I agree, that would be a very useful feature to have.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/generate-platform-diagram.sh (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/generate-platform-diagram.sh:12]: I'm open to the idea of moving this file as a part of a future PR.

**Review by alittley — APPROVED** [review]

**Pull request merged**